src/tests/ecdh.rs

```diff
@@ -469,7 +469,7 @@ fn test_ecc_derive_nist() {
         derive_template.len() as CK_ULONG,
         &mut s_handle,
     );
-    assert_eq!(ret, CKR_MECHANISM_PARAM_INVALID);
+    assert_eq!(ret, CKR_OK);
```

Comment:
I would extend a checking of the result here a bit.

I think there are also outputs in the test vectors that could be checked, but on the first sight, I was not able to decipher what needs to be changed to run the NIST derivation step here
https://github.com/latchset/kryoptic/blob/main/src/tests/ecdh_vectors.rs